.github/workflows/github-actions-build.yml

```diff
@@ -0,0 +1,168 @@
+          export GOPATH=$GITHUB_WORKSPACE/terratest
+          curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+          export PATH=$PATH:$GITHUB_WORKSPACE/terratest/bin
+          cd terratest/src/test
+          dep ensure
+          # grep "FAIL" -A 8 - filter unnecessary logs from the final output but show failing tests
+          #  ; test ${PIPESTATUS[0]} -eq 0 - check if go test was finished success. Without it pipe return exit code from last command.
+          # go test test/pega | grep "FAIL" -A 8 ; test ${PIPESTATUS[0]} -eq 0
+          # go test test/addons | grep "FAIL" -A 8 ; test ${PIPESTATUS[0]} -eq 0
+      
+      - name: run go tests on pega
+        run: |
+          export GOPATH=$GITHUB_WORKSPACE/terratest
+          export PATH=$PATH:$GITHUB_WORKSPACE/terratest/bin
+          cd terratest/src/test
+          go test test/pega | grep "FAIL" -A 8 || true ; test ${PIPESTATUS[0]} -eq 0
+
+      - name: run go tests on addons
+        run: |
+          export GOPATH=$GITHUB_WORKSPACE/terratest
+          export PATH=$PATH:$GITHUB_WORKSPACE/terratest/bin
+          cd terratest/src/test
+          go test test/addons | grep "FAIL" -A 8 || true ; test ${PIPESTATUS[0]} -eq 0
+
+      - name: run go tests on backing services
+        run: |
+          export GOPATH=$GITHUB_WORKSPACE/terratest
+          export PATH=$PATH:$GITHUB_WORKSPACE/terratest/bin
+          cd terratest/src/test
+          go test test/backingservices | grep "FAIL" -A 8 || true ; test ${PIPESTATUS[0]} -eq 0
+
+  run-deploy-job:
+    runs-on: ubuntu-18.04
+    needs: [run-lint-job, run-remark-job, run-go-tests-job]
+    if: (github.event_name != 'pull_request' ) &&  success()
+    steps:
+      - name: Install Helm
+        run: |
+          wget -q ${{ env.HELM_URL }}/${{ env.HELM_TGZ }}
+          tar xzf ${{ env.HELM_TGZ }}
+          ls -l
+          mv linux-amd64/helm /usr/local/bin/helm
+          
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Now load the helm dependencies
+        run: |
+          make dependencies
+
+      - name: before deploy step 1
+        run: |
+          cd $GITHUB_WORKSPACE
+          chmod 777 before_deploy.sh
+          ./before_deploy.sh
+
+      - name: before deploy step 2
+        run: |
+          make examples 
+
+
+      - name: deploy on github releases
+        if: |
+          github.ref_type == 'tag' && startsWith(github.ref, 'refs/tags/v')
+        uses: marvinpinto/action-automatic-releases@latest
+        with:
+          prerelease: false
+          files: |
+            pega-kubernetes-example.tar.gz
+            pega-openshift-example.tar.gz
+            pega-azure-aks-example.tar.gz
+            pega-aws-eks-example.tar.gz
+            pega-google-gke-example.tar.gz
+            pega-pivotal-pks-example.tar.gz
+          repo_token: ${{ secrets.NEW_SECRET }}
+
+      - name: deploy on github pages
```

Comment:
It appears, publish to gh-pages is happening for non tag builds as well. Even on travis too, gh-pages commit is happening for non tag builds, which seems to be un-intentional. 
@dcasavant  Can you please confirm?

Reply:
@kishorv10 I agree, it seems unintentional. We should probably change to tagged builds only for publishing to gh-pages.

Reply:
made the changes to push to gh-pages only for tagged builds